project-base/storefront/components/Basic/ExtendedNextLink/ExtendedNextLink.tsx

```diff
@@ -67,3 +70,22 @@ export const ExtendedNextLink: FC<ExtendedNextLinkProps> = ({
         </NextLink>
     );
 };
+
+const isHrefExternal = (href: string | UrlObject, baseUrl: string) => {
+    const currentHostname = new URL(baseUrl).hostname;
+
+    if (typeof href === 'object') {
+        const linkHostname = href.hostname;
+        if (!linkHostname) {
+            return false;
+        }
+        return currentHostname !== linkHostname;
+    }
+
+    try {
+        return currentHostname !== new URL(href).hostname;
+    } catch (e) {
+        // URL is not valid -> relative pathname -> internal link
```

Comment:
I think the code is self-explanatory. We try to avoid comments, because there is nothing that ages quicker than comments in code.

Reply:
OK I wasn't sure about the clarity, but I also avoid using comments.